### PR TITLE
Migrate federation related flags from e2e-runner to the scenario

### PR DESCRIPTION
**Pull request by krzyzacy**
Moved --up, --down, --federation, --deployment, and partially --save to e2e scenario.

#2829

### CODE REVIEW

**Comment by krzyzacy** [comment]
/unassign
/assign @fejta @madhusudancs 

Basically move those if logic from e2e-runner to some more direct flags, should be easier to configure. PTAL with your eagle eyes so that I won't screw up any federation jobs.

**Comment by shashidharatd** [comment]
Cool, reduces lot of complexity. LGTM from federation jobs perspective.
Probably after this PR, the variable `FEDERATION_CLUSTERS` in all federation jobs is rendered useless. We can clean them up in follow-up PR.
@madhusudancs, What do you think?

**Comment by krzyzacy** [comment]
@shashidharatd the list of zones are not used anywhere? (looks like so..) If that's the case I'll just delete them here

**Comment by shashidharatd** [comment]
Yes @krzyzacy, looks like the variable is not used anywhere else. I would double confirm, once I am in front of system. It's late for me today.

**Comment by krzyzacy** [comment]
@shashidharatd @madhusudancs cleared `FEDERATION_CLUSTERS`, although I don't see how the below comment was achieved before?

Let you take a final look before I merge this.
```
# Recycle control plane.
# We only recycle federation control plane in each run, we don't
# want to recycle the clusters, it's too slow.
# This is accomplished by not setting FEDERATION_CLUSTERS env
# var or setting it to empty string. We set it to empty string
# to explicitly call it out.
```

File: jobs/ci-kubernetes-e2e-gce-federation-serial.env [hunk 1]
```diff
@@ -1,5 +1,4 @@
 ### job-env
-FEDERATION=true
```
Comment (madhusudancs) [inline jobs/ci-kubernetes-e2e-gce-federation-serial.env]: I don't think this should be removed. There are a ton of things in the core repo that depend on this environment variable. For example, see https://github.com/kubernetes/kubernetes/blob/master/hack/e2e-internal/e2e-up.sh#L32.

File: jobs/ci-kubernetes-e2e-gce-federation-serial.env [hunk 2]
```diff
@@ -8,7 +7,6 @@ KUBE_NODE_OS_DISTRIBUTION=gci
 
 # Where the clusters will be created.
 E2E_ZONES=us-central1-a us-central1-b us-central1-f
-FEDERATION_CLUSTERS=us-central1-a us-central1-b us-central1-f
```
Comment (madhusudancs) [inline jobs/ci-kubernetes-e2e-gce-federation-serial.env]: My goal is to replace `E2E_ZONES` with `FEDERATION_CLUSTERS`. The trouble with the current approach is that the cluster's name is dependent on the zone in which it runs. This means we cannot have 2 clusters in the same zone in our tests, which is pretty bad. So I want to move to `FEDERATION_CLUSTERS` and in the transition phase we need both.

File: jobs/ci-kubernetes-e2e-gce-federation-serial.env (hunk 1 above)
Comment (krzyzacy) [inline jobs/ci-kubernetes-e2e-gce-federation-serial.env]: good call! reverted

File: jobs/ci-kubernetes-e2e-gce-federation-serial.env (hunk 2 above)
Comment (krzyzacy) [inline jobs/ci-kubernetes-e2e-gce-federation-serial.env]: ah, I'll leave them there then

File: jobs/ci-kubernetes-e2e-gce-federation-serial.env (hunk 1 above)
Comment (krzyzacy) [inline jobs/ci-kubernetes-e2e-gce-federation-serial.env]: Only cleaned up `FEDERATION_UP`, `FEDERATION_DOWN`, and `JENKINS_FEDERATION_PREFIX`

**Comment by madhusudancs** [comment]
/lgtm

(For federation. Thanks for this PR! It's much cleaner now!)

**Comment by krzyzacy** [comment]
@madhusudancs thanks! will merge tomorrow with a new image in case the world would explode

**Comment by krzyzacy** [comment]
@k8s-bot pull-test-infra-bazel test this

**Pull request merged**